src/components/Equity/Equity.tsx

```diff
@@ -57,6 +57,11 @@ export const Equity = ({ assetId, accountId }: EquityProps) => {
   const totalEquityBalance = useAppSelector(state => selectEquityTotalBalance(state, filter))
   const equityRows = useAppSelector(state => selectAssetEquityItemsByFilter(state, filter))
 
+  const shouldDisplayEquityRows = useMemo(
+    () => equityRows.length > 0 || portfolioLoading,
```

Comment:
Failure to do so means we may end up in a case where portfolio is loaded, but opportunitiesSlice hasn't finished loading yet, resulting in a loss of loading state

```suggestion
    () => equityRows.length > 0 || portfolioLoading || opportunitiesLoading,
```

Reply:
Nice catch